src/vm/corhost.cpp

```diff
@@ -460,10 +460,23 @@ HRESULT CorHost2::ExecuteAssembly(DWORD dwAppDomainId,
             arguments->SetAt(i, argument);
         }
 
-        DWORD retval = pAssembly->ExecuteMainMethod(&arguments, TRUE /* waitForOtherThreads */);
-        if (pReturnValue)
+        if(CLRConfig::GetConfigValue(CLRConfig::INTERNAL_Corhost_Swallow_Uncaught_Exceptions))
         {
-            *pReturnValue = retval;
+            EX_TRY
```

Comment:
These macros  are rightly coupled with the rest of CoreCLR. They do not simulate well what code outside CoreCLR.dll catching the exception.

Reply:
They were at least able to suppress the event log which is triggered by the unhandled exception handler.  Which allowed testing the new mechanism to enable logging caught exceptions.

Do you have a different recommendation?  Creating a new host for this test seems like overkill....

Reply:
I guess I could place it under `#ifdef PLATFORM_WINDOWS` and use `__try`/`__catch` if you think that would be better.

Reply:
> Do you have a different recommendation?

Would it make sense to wait for https://github.com/dotnet/core-setup/pull/5336 ? Creating new hosts for testing should be simple with that, I think.

Reply:
Given @pakrym & @davidfowl's priority on this, I am thinking we should merge as is, and open an issue to fix after https://github.com/dotnet/core-setup/pull/5336 is implemented.

I guess the alternative, would be to put the test in a separate PR and remove the environment variables from this PR.

Reply:
I would be ok with not having tests included in the PR - if waiting for dotnet/core-setup#5336  makes sense.

Reply:
I guess one other approach would be to 
+ keep the EX_TRY/EX_CATCH_HR, 
+ add another property/key pair (SWALLOW_UNCAUGHT_EXCEPTIONS) to allow it to be enabled
+ recommend ASP.NET's host use this mechanism.

Reply:
@vitek-karas Do you have a preference?

Reply:
I would only use the property as already implemented and merge that.
I can live with it not having tests for now.

That said, writing a one off test which manually initializes coreclr and invokes a super simple main should not be that hard (we can hardcode TPA and the rest of the needed properties). I know we don't really have hosting API tests, but that doesn't mean we should not try to add them :-)